### ISSUE TITLE
Pivot Table requires results to be loaded before Summarize fields can be changed

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#26467]
**Describe the bug**
The Summarize group-by fields are nonfunctional on Pivot Table until results are loaded making the UI feel broken or slow.
Since Pivot Table visualization is a lot slower than any other visualization, then it is easy to encounter this problem. #21688

Workaround: Wait for results to be loaded, then changes to the summarize group-by works.

**To Reproduce**
1. Create an intentionally slow query - save question as "Slow query"
```
-- postgres
select 1 as ID, 'cat' as VAL, 'cat2' as VAL2, pg_sleep(10)
-- mysql
select 1 as ID, 'cat' as VAL, 'cat2' as VAL2, sleep(10)
```
2. Question > Saved Questions > "Slow query" - summarize Count grouped by VAL
![image](https://user-images.githubusercontent.com/1447303/201519380-5c3dfad0-e4f2-4902-a6ec-1b475e7cf1d7.png)
3. Change visualization to Pivot Table - wait for it to finish load results
4. Save question and do a browser refresh
5. While results are being loaded, try changing the summarize group-by fields - nothing happens

```
Uncaught (in promise) TypeError: Cannot read properties of null (reading '0')
    at e (icepick.min.js:1:2152)
    at core.js:276:22
    at u (runtime.js:63:40)
    at Generator._invoke (runtime.js:293:22)
    at Generator.next (runtime.js:118:21)
    at Vt (native.js:149:63)
    at a (native.js:149:63)
    at native.js:149:63
    at new Promise (<anonymous>)
    at native.js:149:63
```

https://user-images.githubusercontent.com/1447303/201519838-9473f8eb-e173-4626-a220-47b88b78a75a.mp4

**Expected behavior**
Should be possible the change summarized fields before results are loaded, since metadata is already available.

**Information about your Metabase Installation:**
Tested 0.44.6

**Comment by npretto** [comment]
This seems to have been fixed in 46.0, but was still broken in 45.4

**Comment by npretto** [comment]
This was closed with https://github.com/metabase/metabase/pull/28945/
